akka-http-core/src/main/scala/akka/http/javadsl/OutgoingConnectionBuilder.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.annotation.DoNotInherit
+import akka.event.LoggingAdapter
+import akka.http.javadsl.model.HttpRequest
+import akka.http.javadsl.model.HttpResponse
+import akka.http.javadsl.settings.ClientConnectionSettings
+import akka.http.scaladsl.OutgoingConnectionBuilder
+import akka.stream.javadsl.Flow
+
+/**
+ * Builder for setting up a flow that will create one single connection per materialization to the specified host.
+ * When customization is done, the flow is created using [[#connectionFlow()]].
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+trait OutgoingConnectionBuilder {
+  /**
+   * Change which host flows built with this builder connects to
+   */
+  def toHost(host: String): OutgoingConnectionBuilder
```

Comment:
I think reusing is not the main reason but more a possibility to override the host after the fact if you want to abstract setting some features of the builder in user-code which would otherwise require to do the host thing first.

But in general, I agree, and it would also be possible to remove this one completely.